lutris/installer/silient_install.py

```diff
@@ -0,0 +1,219 @@
+        else:
+            self._print(self.commandline, _("Waiting for response from %s" % settings.SITE_URL))
+            logger.info(_("Waiting for response from %s" % settings.SITE_URL))
+            jobs.AsyncCall(
+                interpreter.fetch_script,
+                self.on_scripts_obtained,
+                self.game_slug,
+                self.revision
+            )
+
+    def on_scripts_obtained(self, scripts, _error=None):
+        if not scripts:
+            self._print(self.commandline, "No install script found")
+            logger.error("No install script found")
+            return
+
+        if isinstance(scripts, list):  # if scripts is a list longer than one element
+            if len(scripts) != 1:
+                self._print(self.commandline, "Please provide single installer")
+                logger.error("No install script found")
+                return
+            self.script = scripts[0]
+        else:
+            self.script = scripts
+
+        self.validate_scripts()
+        self.prepare_install(self.script)
+
+    def validate_scripts(self):
+        """Auto-fixes some script aspects and checks for mandatory fields"""
+
+        for item in ["description", "notes"]:
+            self.script[item] = self.script.get(item) or ""
+        for item in ["name", "runner", "version"]:
+            if item not in self.script:
+                self._print(self.commandline, _("Invalid script: %s" % self.script))
+                raise ScriptingError('Missing field "%s" in install script' % item)
+
+    def prepare_install(self, script):
+
+        install_script = script
+        if not install_script:
+            raise ValueError("Could not find script %s" % install_script)
+        try:
+            self.interpreter = interpreter.ScriptInterpreter(install_script, self)
+        except MissingGameDependency as ex:
+            # call recursive dependencies
+            SilenInstall(
+                game_slug=ex.slug
+            )
+
+        self.select_install_folder()
+
+    def select_install_folder(self):
+        """Stage where we select the install directory."""
+        if self.interpreter.creates_game_folder:
+            if self.install_path is None:
+                self.install_path = self.interpreter.get_default_target()
+
+            self.interpreter.target_path = self.install_path
+            self._print(self.commandline, _("install Folder %s" % self.interpreter.target_path))
+            logger.info("No install script found")
+
+        try:
+            self.interpreter.check_runner_install()
+        except ScriptingError as ex:
+            self._print(self.commandline, ex.__str__)
+            logger.error(ex.__str__)
+            return
+
+    def set_status(self, text):
+        self._print(self.commandline, text)
+
+    # required by interpreter
+    def clean_widgets(self):
+        pass
+
+    # required by interpreter
+    def add_spinner(self):
```

Comment:
Required? This SilentInstall class does not extend or inherit any class, so where is this requirement coming from?